### ISSUE TITLE
Function that returns the first visible row (or all visible rows) of a Grid

### ISSUE THREAD

**Issue by aiolos01** [vaadin/framework#9708]
With the table we were able to get the first line that is currently visible on the client's UI. Grid does not have this important functionality. 
There should be a way to get this information (or alternatively all visible rows) as it can be useful in any number of situations.

**Comment by aiolos01** [comment]
Still no way of doing this. I believe it's a very useful feature that should be implemented.

**Comment by TatuLund** [comment]
This is actually possible by using grid.getDataCommunicator().fetchItemsWithRange(0,1).get(0);, see also this another ticket: https://github.com/vaadin/framework/issues/10935

**Comment by aiolos01** [comment]
Thanks for your answer. Maybe I didn't make it clear enough. I'm not looking for the first line of a grid but the first currently visible line after the user has scrolled the grid. This was available on the table component.
This is very useful when the user deletes a line and the grid has to be redrawn. If the grid is scrolled back to the first line then the user needs to scroll back to where he was manually.

**Comment by TatuLund** [comment]
> When the user deletes a line and the grid has to be redrawn. If the grid is scrolled back to the first line then the user needs to scroll back to where he was manually.

Deleting a row and calling DataProvider.refreshAll() does not scroll back the Grid.

However if you need to track scroll position and set it programmatically, one option is to use this add-on  https://vaadin.com/directory/component/gridscrollextension-add-on

**Comment by aiolos01** [comment]
You're right about the grid. Until very recently I used the legacy v.7 grid. I switched to v.8 a few days ago and hadn't tested that particular functionality. By chance I did that yesterday just before seeing your comment. Thanks for confirming this.

The add-on is very useful too. I'll be sure to test it. Thank you,
